### PR TITLE
Upgrade RESTEasy from 4.6.1.Final to 4.7.0.Final

### LINKED ISSUE #18622: Upgrade RESTEasy from 4.6.1.Final to 4.7.0.Final
### Description

Release Notes for 4.7.0.Final: https://issues.redhat.com/secure/ReleaseNote.jspa?projectId=12310560&version=12353153

### Implementation ideas

_No response_

### PR DESCRIPTION
**Pull request by jamezp**
Resolves #18622

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -291,6 +281,17 @@
...
+            <!-- RESTEasy dependencies, imported as a BOM -->
+            <!-- As RESTEasy also imports Jackson, we need to import its BOM after the Jackson BOM -->
+            <!-- As RESTEasy also imports Vert.x transitively, we need to import its BOM after the Vert.x BOM -->
+            <dependency>
+                <groupId>org.jboss.resteasy</groupId>
+                <artifactId>resteasy-bom</artifactId>
+                <version>${resteasy.version}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
```
Comment (gsmet) [inline bom/application/pom.xml:293]: Why did you move the BOM? The BOM order is important so I wouldn't move it except if we have a good reason.

**Review by gsmet — CHANGES_REQUESTED** [review]
Thanks. I added a small comment.

File: bom/application/pom.xml (hunk 1 above)
Comment (jamezp) [reply bom/application/pom.xml:293]: There was a conflict with the version of Vert.x the RESTEasy BOM was importing. It was causing compile errors. I just moved it below the Vert.x import.

**Review by asoldano — APPROVED** [review]

File: bom/application/pom.xml (hunk 1 above)
Comment (gsmet) [reply bom/application/pom.xml:293]: Why is the RESTEasy BOM including a version of Vert.x? This doesn't seem like a normal behavior.

**Review by gsmet — CHANGES_REQUESTED** [review]
Let's sort out this ordering issue before thinking about merging this PR.

**Comment by jamezp** [comment]
> Let's sort out this ordering issue before thinking about merging this PR.

I'm not too sure what could be done to sort it out short of using the same version of Vert.x in Quarkus and RESTEasy. Even then it just hides it until it diverges again.

File: bom/application/pom.xml (hunk 1 above)
Comment (stuartwdouglas) [reply bom/application/pom.xml:293]: RESTEasy has a Vert.x module (that is different to the Vert.x support we provide).

Comment (gsmet) [reply bom/application/pom.xml:293]: Yes sure but it shouldn't include a version for Vert.x artifacts.

I put the BOM back where it was and I don't have any compilation issue.

I'll push to a branch in my fork to see what CI has to say.

Comment (gsmet) [reply bom/application/pom.xml:293]: Also I made a diff of the effective POM and there's no difference except for the position. BOMs are a sensible matter so I prefer not changing it if we can avoid it.

**Comment by jamezp** [comment]
@gsmet I think that's because of this fix https://github.com/resteasy/Resteasy/pull/2856/files#diff-804f7d09a86dc42c94770466e216189fa8816942d41764cb4d31a69b2013f37dR30-R31. If you use 4.7.0.Beta1 which doesn't include that and uses Vert.x 4.0.3 you'll see a compile error in `extensions/vertx-http/runtime`.

I will admittedly do not understand how the `dependency:tree` shows the right version with Final, but not Beta.

Anyway whatever you prefer. If an upgrade to Vert.x breaks it in the future we'll know how to fix it :)

File: bom/application/pom.xml [hunk 2]
```diff
@@ -256,6 +256,7 @@
 
             <!-- RESTEasy dependencies, imported as a BOM -->
             <!-- As RESTEasy also imports Jackson, we need to import its BOM after the Jackson BOM -->
+            <!-- As RESTEasy also imports Vert.x transitively, we need to import its BOM after the Vert.x BOM -->
```
Comment (jamezp) [inline bom/application/pom.xml:259]: This could be removed now since we didn't move the import.

Comment (gsmet) [reply bom/application/pom.xml:259]: Yeah, I'll clean everything if I can get CI to pass. If not I'll dig a bit more.

**Review by gsmet — APPROVED** [review]
Things are looking good. I squashed the commits.

**Comment by famod** [comment]
CI is happy, so let's merge.

I'll follow up with re-enabling those rest-client TCK tests that had to be disabled for Java 16.
